### DISCUSSION TITLE
Configuration of highlight when using superfences

### DISCUSSION THREAD

**Discussion by TheStaticTurtle** [facelessuser/pymdown-extensions#1653] [Q&A] [answered]
Hello,

I just spent 3h trying to figure out why I could configure `pymdownx.highlight` when only adding `pymdownx.superfences` to the extension list. Turns out you need to add both to the extension list.

Is it written somewhere that you actually need both in the extension list in order to configure it? Because if not, it's extremely confusing when a new user add `pymdownx.superfences` and you expect to be able to configure the `pymdownx.highlight` module.

**Comment by facelessuser** [comment, marked as answer]
Pymdown Extensions has two extensions that use highlighting: SuperFences and InlineHilite. They are both built on Highlight which does the actual formatting of the code to syntax highlighted code blocks (or blocks ready for JS highlighting if not using Pygments).

SuperFences allows you to configure the fenced code syntax and behavior just as InlineHilite has options specifically for controlling its syntax and behavior. Highlight controls the formatting of the code blocks.

It is noted that SuperFences uses `Highlight` under the hood and that all code highlighting features are controlled by Highlight:

- https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#code-highlighting.
- https://facelessuser.github.io/pymdown-extensions/extensions/inlinehilite/#code-highlighting

If using SuperFences or InlineHIlite, Highlight is automatically included and used. You do not have to manually include it **unless** you specifically want to configure it.

While an argument could be made that this should somehow be made more clear, I'm open to suggestions on how to do that. I'll be honest, I don't admit to being the best document writer, but I do try to make things as clear as I can.

**Reply by TheStaticTurtle** [reply]
Hi, thanks for the response
> If using SuperFences or InlineHIlite, Highlight is automatically included and used. You do not have to manually include it unless you specifically want to configure it.

That was my point, reading the doc it's pretty clear that SuperFences  or InlineHilite uses Highlight in the background, but it isn't that you need to include Highlight manually to configure it 

I'm not the best doc writer myself, but you cloud add a note like this ``Note: you need to include `pymdownx.highlight` in the extensions list in order to be able to configure it``

**Reply by facelessuser** [reply]
Yeah, I mean there is no way to configure it unless you include it, but we could definitely add a note to clarify. Can you open an issue?
